### PR TITLE
Resolve scopes relative to import map base

### PR DESCRIPTION
**Pull request by guybedford**
This implements https://github.com/systemjs/systemjs/issues/1968 in resolving scopes relative to the baseUrl, as was implemented in the import maps spec update.

As a major change this will form a SystemJS v5, so it's worth thinking about if we have any other major changes we'd like to add to the mix before merging and releasing here.

### CODE REVIEW

File: src/common.js [hunk 1]
```diff
@@ -134,7 +134,7 @@ export function parseImportMap (json, baseUrl) {
       let resolvedScopeName = resolveUrl(scopeName, baseUrl);
```
Comment (joeldenning) [inline src/common.js]: Can we remove the entire resolvedScopeName variable? Looks like it’s dead/unused?

File: src/common.js [hunk 2]
```diff
@@ -156,8 +156,7 @@ function applyPackages (id, packages) {
   const pkgName = getMatch(id, packages);
   if (pkgName) {
     const pkg = packages[pkgName];
-    if (pkg === null)
-
+    if (pkg === null) return;
```
Comment (joeldenning) [inline src/common.js]: Is this fixing an existing bug? Looks like we just had an if statement without a block afterwards?

File: src/common.js (hunk 1 above)
Comment (joeldenning) [reply src/common.js]: Oops I misread this — disregard 😀

File: src/common.js (hunk 2 above)
Comment (guybedford) [reply src/common.js]: Yup! The validation was never applying :astonished:

**Pull request merged**